### ISSUE TITLE
Local svg icons for sidebar

### ISSUE THREAD

**Issue by groblus** [cosmocode/dokuwiki-template-sprintdoc#19]
Hi, I've got some issues with using local SVG icons. I have no problem for using icons from the MDI set, but when I want to switch to my own SVG files - then default icon appears.
As an an example:
```
====== Head1 @test.svg ======
{{test.svg}}
====== Head2 @alert-octagram.svg ======
```
And icon located inside wiki ({{test.svg}}) renders correctly.
alert-octagram.svg renders correctly as Head2 icon, but Head1 gets a default one. 
Perhaps my svg files should be located in some special directory or perhaps I'm missing something obvious.
Regards

**Comment by diegojancic** [comment]
**TL;DR;:**
1) Place your svgs in `/lib/tpl/sprintdoc/img`, in lowercase
2) Specify the icons as `====== Title @icon-name.svg# ======` (with # at the end)

I was able to find a workaround. First, the icons don't load because of the following line (in file `svg.php`):

`if(auth_quickaclcheck($svg) < AUTH_READ) $this->abort(403);`

For some reason, the ACL check is returning AUTH_NONE and the script returns HTTP 403.

The workaround is the following.  In the directory: `/lib/tpl/sprintdoc/img`, upload your SVG files. The file name must be in all lowercase. The reason is that Linux is case sensitive and the request from the browser is made using lowercase, so the files need to match that.

Also, the script svg.php makes some transformations (not sure why). So if you're file doesn't look fine, you might need to skip those transformations. To do so, instead of using:

`====== Title @icon-name.svg ======`

Use:

`====== Title @icon-name.svg# ======`

The reason for that is that the request is made as follows:

http://domain.com/lib/tpl/sprintdoc/svg.php?svg=icon-name.svg&e=1

That e=1 at the end is what tells the script to apply the transformation. By adding the # symbol, we are 'commenting out' that parameter, resulting in this:

http://domain.com/lib/tpl/sprintdoc/svg.php?svg=icon-name.svg#&e=1

**Comment by annda** [comment]
Fixed in the latest release, the suggested workaround is no longer necessary. However, the icons must be **publicly accessible**, that is not protected by any ACL rules.
